### PR TITLE
Linux

### PR DESCRIPTION
**Pull request by dsemenets**
Linux and bootscript recipes

### CODE REVIEW

File: recipes-kernel/linux/files/set-cmdline.cfg [hunk 1]
```diff
@@ -0,0 +1 @@
+CONFIG_CMDLINE="rw console=hvc0 clk_ignore_unused earlycon=xenboot debug root=/dev/ram0"
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/set-cmdline.cfg]: where this will go? cmdline.txt at RPI5 bootfs?

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: GPL-2.0
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Why is this file needed? Now we add all Xen cfg with boot scripts.

File: recipes-kernel/linux/files/set-cmdline.cfg (hunk 1 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/set-cmdline.cfg]: No. Linux .config

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 2 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: I don't think it is good idea. This noticeable reduce readability.
I prefer modify some items in boot script.

File: recipes-kernel/linux/files/set-cmdline.cfg (hunk 1 above)
Comment (oleksiimoisieiev) [reply recipes-kernel/linux/files/set-cmdline.cfg]: have you checked if cmdline.txt override cmdline on Linux dtb or not? That's important i think

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+			#size-cells = <0x2>;
+			compatible = "xen,domain";
+			memory = <0 524288>;
+			cpus = <1>;
+			xen,static-mem = <0x0 0x40000000 0x0 0x20000000>;
+			direct-map;
+			vpl011 = <0x1>;
+			xen,enhanced = "enabled";
+		};
+		DomU1 {
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: domU?

File: recipes-kernel/linux/files/set-cmdline.cfg [hunk 4]
```diff
@@ -0,0 +1 @@
+CONFIG_CMDLINE="console=ttyAMA0 earlycon=xen earlyprintk=xen clk_ignore_unused"
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/files/set-cmdline.cfg]: again about CMD line? Is it used? I probably can be overriden with cmdline.txt

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+COMPATIBLE_MACHINE:raspberrypi5 = "(raspberrypi5)"
+
+DTB_NAME = "${RPI_SOC_FAMILY}-${MACHINE}-xen"
+RPI_KERNEL_DEVICETREE:append = " broadcom/${DTB_NAME}.dtb"
+
+SRC_URI:append = " \
+    ${@bb.utils.contains("DISTRO_FEATURES", "xen", "file://xen-console.cfg", "", d)} \
+    file://set-cmdline.cfg \
+    file://${DTB_NAME}.dts;subdir=git/arch/${ARCH}/boot/dts/broadcom \
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: Have you checked if this file builds? That's because I don't see any changes to the Makefile which will add this to the build

Comment (dsemenets) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: I'm checking this every day. It works

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 3 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: I have changed this

File: recipes-kernel/linux/files/set-cmdline.cfg (hunk 4 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/set-cmdline.cfg]: Kernel adds this to cmd line got from bootloader. A'll remove this

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+*/
+#include "bcm2712-rpi-5-b.dts"
+
+/ {
+	chosen {
+		#address-cells = <0x2>;
+		#size-cells = <0x2>;
+		xen,xen-bootargs = "console=dtuart dtuart=/soc/serial@7d001000 dom0_mem=256M dom0_max_vcpus=1 xsm=flask flask=permissive";
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: One comment - change dom0_mem=256M to dom0_mem=128M

**Comment by GrygiriiS** [comment]
LGTM

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+fdt mknod /chosen/domU0 module@@@DOMU0_DTB_ADDR@@
+fdt set /chosen/domU0/module@@@DOMU0_DTB_ADDR@@ compatible  "multiboot,device-tree" "multiboot,module"
+fatload @@BOOT_MEDIA@@ 0 @@DOMU0_DTB_ADDR@@ @@DOMU0_DTB@@
+fdt set /chosen/domU0/module@@@DOMU0_DTB_ADDR@@ reg <0x0 @@DOMU0_DTB_ADDR@@ 0x0 0x${filesize} >
+
+setenv fdt_high 0xffffffffffffffff
+
+fdt print /chosen
+printenv
```
Comment (oleksiimoisieiev) [inline recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: please no prints

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend [hunk 8]
```diff
@@ -0,0 +1,18 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+COMPATIBLE_MACHINE:raspberrypi5 = "(raspberrypi5)"
+
+DOMU_DTB_NAME = "${RPI_SOC_FAMILY}-${MACHINE}-domu"
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: what is domu-dtb? Are we going to have domu?

Comment (oleksiimoisieiev) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: It is DOMD_DTB_NAME in moulin

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    broadcom/${DOMU_DTB_NAME}.dtb \
+    broadcom/${XEN_DTB_NAME}.dtb \
+"
+
+KERNEL_IMAGETYPES:append = " Image.gz"
+
+SRC_URI:append = " \
+    ${@bb.utils.contains("DISTRO_FEATURES", "xen", "file://xen-config.cfg", "", d)} \
+    file://${DOMU_DTB_NAME}.dts;subdir=git/arch/${ARCH}/boot/dts/broadcom \
+    file://${XEN_DTB_NAME}.dts;subdir=git/arch/${ARCH}/boot/dts/broadcom \
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/linux-raspberrypi_6.6.bbappend:16]: Still not clear to me how do you build it without adding to the makefile. 
I can tell why it works: because, according to the yaml we have the following line in deploy:
       - "tmp/deploy/images/%{RPI_MACHINE}/%{XT_XEN_DTB_NAME}"
        - "tmp/deploy/images/%{RPI_MACHINE}/bcm2712-rpi-5-b.dtb"
So rpi5 using bcm2712-rpi-5-b.dtb to boot and all changes you have done in xen are not applied

**Review by oleksiimoisieiev — CHANGES_REQUESTED** [review]

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 6 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Ok

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in (hunk 7 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: Ok

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 9 above)
Comment (dsemenets) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend:16]: But sources don't have tmp/deploy/images/%{RPI_MACHINE}/%{XT_XEN_DTB_NAME}. This should be built some how.
I carefully checked. All changes are applied. Boot script can't changes on vanilla bcm2712-rpi-5-b.dtb. This will cause error messages bu I don't see any error message and result chosen node looks good. Also %{XT_XEN_DTB_NAME} provides any correction to vanilla dtb like delete some nodes and adds xen,passthrough property to another nodes. Those changes are present.

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+fatload @@BOOT_MEDIA@@ 0 @@XENPOLICY_IMG_ADDR@@ @@XENPOLICY_IMAGE@@
+fdt set /chosen/modules/module@2 reg <0x0 @@XENPOLICY_IMG_ADDR@@ 0x0 0x${filesize} >
+
+fatload @@BOOT_MEDIA@@ 0 @@DOM0_IMG_ADDR@@ @@DOM0_IMAGE@@
+fdt set /chosen/dom0 reg <0x0 @@DOM0_IMG_ADDR@@ 0x0 0x${filesize} >
+
+fdt mknod /chosen/domU0 module@@@DOMU0_IMG_ADDR@@
+fdt set /chosen/domU0/module@@@DOMU0_IMG_ADDR@@ compatible  "multiboot,kernel" "multiboot,module"
+fdt set /chosen/domU0/module@@@DOMU0_IMG_ADDR@@ bootargs "console=ttyAMA0 earlycon=xen earlyprintk=xen clk_ignore_unused root=/dev/mmcblk0p2 rootfstype=ext4 rootwait"
+fatload @@BOOT_MEDIA@@ 0 @@DOMU0_IMG_ADDR@@ @@DOMU0_IMAGE@@
```
Comment (oleksiimoisieiev) [inline recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: Domu?

**Review by oleksiimoisieiev — CHANGES_REQUESTED** [review]

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in (hunk 10 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: DOMD

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 8 above)
Comment (dsemenets) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: Now it is DOMD

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+		xen,dom0-bootargs = "console=hvc0 earlycon=xen earlyprintk=xen clk_ignore_unused root=/dev/ram0";
+		modules {
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: This is incorrect as per Xen ARM booting.txt

- no need to have "module" node fro xen itself
- use correct compats only "multiboot,module" + [ "multiboot,kernel" | "multiboot,ramdisk" | "xen,xsm-policy"] 
- I don't know what is "xen-policy", so can't tell if it belongs to XEN or should belong to Domain. Most probably, it should belong to domain.

as result boot script is also incorrect.

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+DOMD_DTB ?= "${RPI_SOC_FAMILY}-${MACHINE}-domd.dtb"
+DOMD_DTB_ADDR = "0x1c00000"
+XEN_IMAGE ?= "xen"
+XEN_IMG_ADDR ?= "0x4000000"
+XENPOLICY_IMAGE ?= "xenpolicy"
+XENPOLICY_IMG_ADDR ?= "0x4150000"
+XEN_DTB ?= "${RPI_SOC_FAMILY}-${MACHINE}-xen.dtb"
+XEN_DTB_ADDR ?= "0x4200000"
+UBOOT_BOOT_SCRIPT ?= "boot.scr"
+
```
Comment (GrygiriiS) [inline recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:33]: Above should have comment section describing how much memory in Mb allocated for each component

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in (hunk 10 above)
Comment (GrygiriiS) [reply recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: From Xen point of view they are all DomU and it also follows Xen docs

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        -e 's/@@XEN_DTB_ADDR@@/${XEN_DTB_ADDR}/g' \
+        -e 's/@@XENPOLICY_IMAGE@@/${XENPOLICY_IMAGE}/g' \
+        -e 's/@@XENPOLICY_IMG_ADDR@@/${XENPOLICY_IMG_ADDR}/g' \
+        -e 's/@@DOMD_DTB@@/${DOMD_DTB}/g' \
+        -e 's/@@DOMD_DTB_ADDR@@/${DOMD_DTB_ADDR}/g' \
+        -e 's/@@DOMD_IMAGE@@/${DOMD_IMAGE}/g' \
+        -e 's/@@DOMD_IMG_ADDR@@/${DOMD_IMG_ADDR}/g' \
+        "${WORKDIR}/${TEMPLATE_FILE}" > "${WORKDIR}/boot.cmd"
+
+    mkimage -A ${UBOOT_ARCH} -T script -C none -n "Boot script" -d "${WORKDIR}/boot.cmd" ${UBOOT_BOOT_SCRIPT}
```
Comment (GrygiriiS) [inline recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb]: Please add boot.cmd to deploy process.

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+inherit kernel-arch deploy nopackages
```
Comment (lorc) [inline recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:53]: Do you really need `kernel-arch`?

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+do_deploy() {
+    install -d ${DEPLOYDIR}
+    install -m 0644 ${UBOOT_BOOT_SCRIPT} ${DEPLOYDIR}
+}
+
+addtask do_deploy after do_compile before do_build
+
+PROVIDES += "u-boot-default-script"
```
Comment (lorc) [inline recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:63]: Do you really need `PROVIDES`? Where it is used?

File: recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch [hunk 16]
```diff
@@ -0,0 +1,50 @@
+From 95866cace6d5e1dd6e810f18e12969473fad2aa2 Mon Sep 17 00:00:00 2001
+From: Grygorii Strashko <grygorii_strashko@epam.com>
+Date: Wed, 24 Apr 2024 15:10:10 +0300
+Subject: [PATCH] drivers: mmc: host: sdhci-brcmstb: fix no pinctrl case
+
+In no punctrl case, when pinctrl are set by u-boot for example, driver
```
Comment (lorc) [inline recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch]: punctrl?

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+		};
+		domU0 {
+			compatible = "xen,domain";
+			#address-cells = <0x2>;
+			#size-cells = <0x2>;
+			memory = <0x0 0x80000>;
+			vpl011 = <0x1>;
+			direct-map;
+			cpus = <0x1>;
+			xen,static-mem = <0x0 0x50000000 0x0 0x20000000>;
```
Comment (lorc) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts:28]: Does DomU really needs static mem?

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 9 above)
Comment (lorc) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend:16]: I believe this is yocto magic. Check `kernel-devicetree.bbclass` for more info and note `RPI_KERNEL_DEVICETREE` variable.

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 17 above)
Comment (GrygiriiS) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts:28]: Yep. No dma otherwise and no storage/net. That's why we have to play with dom0less boot :(

FYI. https://github.com/xen-troops/meta-xt-rpi5/wiki/RPI-5-Xen-Dom0less-boot-exercise

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 9 above)
Comment (oleksiimoisieiev) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend:16]: yep. already discussed it with @dsemenets .

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+&sdio1          { xen,passthrough; };
+&gio_aon        { xen,passthrough; };
+&rp1_eth        { xen,passthrough; };
+&pcie2          { xen,passthrough; };
+&rp1_gpio       { xen,passthrough; };
+&rp1_clocks     { xen,passthrough; };
+&bcm_reset      { xen,passthrough; };
+&pcie_rescal    { xen,passthrough; };
+&mip0           { xen,passthrough; };
+/delete-node/   &_uart2;
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Add HACK description from https://raw.githubusercontent.com/wiki/xen-troops/meta-xt-rpi5/files/xen_dom0less/ib_boot_extra.source

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 8 above)
Comment (GrygiriiS) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: Summary replace DOMU0 with DOMD

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 14 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:53]: Yes. This recipe can't be compiled without kernel-arch. This module setup UBOOT_ARCH variable used in mkimage argiments

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 15 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:63]: u-boot recipe in meta-raspberrypi depends on u-boot-default-script

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 18 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Added

File: recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch (hunk 16 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch]: Fixed

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 13 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb]: Added

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 12 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:33]: Added

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 11 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Fixed

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 8 above)
Comment (dsemenets) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: Replaced

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 14 above)
Comment (lorc) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:53]: This is just a u-boot script. There is no compilation per se. On other hand, "kenrel-arch" does basically this:

```
export ARCH = "${@map_kernel_arch(d.getVar('TARGET_ARCH'), d)}"
export UBOOT_ARCH = "${@map_uboot_arch(d.getVar('ARCH'), d)}"
```

Why do you need this variables if you are not compiling anything?

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts [hunk 19]
```diff
@@ -0,0 +1,287 @@
...
+			aspm-no-l0s;
+			brcm,vdm-qos-map = <0xbbaa9888>;
+			brcm,enable-mps-rcb;
+			brcm,enable-l1ss;
+			dma-ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0x400000
+				      0x43000000 0x10 0x0 0x0 0x0 0x10 0x0>;
+			ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0xfffffffc
+				  0x43000000 0x4 0x0 0x1c 0x0 0x3 0x0>;
+			interrupt-parent = <&gic>;
+//			msi-parent = <&mip0>;
```
Comment (lorc) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts]: Please drop commented out code.

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 17 above)
Comment (lorc) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts:28]: Ah, this is DomD. Got it.

File: recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch [hunk 20]
```diff
@@ -0,0 +1,50 @@
+From 95866cace6d5e1dd6e810f18e12969473fad2aa2 Mon Sep 17 00:00:00 2001
```
Comment (oleksiimoisieiev) [inline recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch:1]: devixe-> device in commit description

**Comment by oleksiimoisieiev** [comment]
+1 with a small comment. `Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>`

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts [hunk 21]
```diff
@@ -0,0 +1,287 @@
...
+			aspm-no-l0s;
+			brcm,vdm-qos-map = <0xbbaa9888>;
+			brcm,enable-mps-rcb;
+			brcm,enable-l1ss;
+			dma-ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0x400000
+				      0x43000000 0x10 0x0 0x0 0x0 0x10 0x0>;
+			ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0xfffffffc
+				  0x43000000 0x4 0x0 0x1c 0x0 0x3 0x0>;
+			interrupt-parent = <&gic>;
+//			msi-parent = <&mip0>;
```
Comment (lorc) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:162]: Please no commented out code.

Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:162]: @GrygiriiS asked do not remove it

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts (hunk 19 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts]: Look at 2 lines above.
This script use mkimage to create bootscript. mkimage needs UBOOT_ARCH as argument.
I can take needed value manually or use kernel-arch

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts (hunk 21 above)
Comment (lorc) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:162]: It deserves a comment then. Something like "HACK: Actual MSI parent is &mip but due to.... we temporary use different parent"

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+		#size-cells = <0x2>;
+		xen,xen-bootargs = "console=dtuart dtuart=/soc/serial@7d001000 dom0_mem=128M dom0_max_vcpus=1 xsm=flask flask=permissive";
+		xen,dom0-bootargs = "console=hvc0 earlycon=xen earlyprintk=xen clk_ignore_unused root=/dev/ram0";
+		xenpolicy {
+			compatible = "xen,xsm-policy", "xen,multiboot-module";
+		};
+		dom0 {
+			compatible = "xen,linux-zimage", "xen,multiboot-module", "multiboot,module";
+		};
+		domU0 {
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: We agreed you rename domU0 to domD here also.

File: recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in (hunk 7 above)
Comment (GrygiriiS) [reply recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in]: Prints must stay for now! It's very important for current dev stage

File: recipes-kernel/linux/linux-raspberrypi_6.6.bbappend (hunk 8 above)
Comment (GrygiriiS) [reply recipes-kernel/linux/linux-raspberrypi_6.6.bbappend]: not all. I've added comment.

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+#include "bcm2712-rpi-5-b.dts"
+
+/ {
+	chosen {
+		#address-cells = <0x2>;
+		#size-cells = <0x2>;
+		xen,xen-bootargs = "console=dtuart dtuart=/soc/serial@7d001000 dom0_mem=128M dom0_max_vcpus=1 xsm=flask flask=permissive";
+		xen,dom0-bootargs = "console=hvc0 earlycon=xen earlyprintk=xen clk_ignore_unused root=/dev/ram0";
+		xenpolicy {
+			compatible = "xen,xsm-policy", "xen,multiboot-module";
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: This should be

`			compatible = "xen,xsm-policy", ""multiboot,module";
`

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+		};
+		dom0 {
+			compatible = "xen,linux-zimage", "xen,multiboot-module", "multiboot,module";
```
Comment (GrygiriiS) [inline recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: This should be:
`			compatible = ""multiboot,kernel", "multiboot,module";
`

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts (hunk 21 above)
Comment (GrygiriiS) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:162]: You can  get description from: https://github.com/xen-troops/meta-xt-rpi5/wiki/RPI-5-Xen-Dom0less-boot-exercise#pci2-msi-is-not-xen-compatible

```
"HACK: But the mip0: msi-controller@130000 doesn't have IRQs defined in the standard way and instead uses custom properties: brcm,msi-base-spi to define GIC IRQ offset, and brcm,msi-num-spis to define number of GIC IRQs to use. The Xen doesn't understand this and so can't map interrupts to DomU in case of 'xen,passthrough'.
The attempt to add correct interrupts property to the mip0: msi-controller@130000 has failed - no IRQs received by macb 1f00100000.ethernet. Reason unknown."

```

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 22 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: Done

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 23 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: done

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts (hunk 24 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-xen.dts]: done

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts (hunk 21 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts:162]: Comment added

**Comment by GrygiriiS** [comment]
LGTM

File: recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch (hunk 20 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/0001-drivers-mmc-host-sdhci-brcmstb-fix-no-pinctrl-case.patch:1]: Fixed

File: recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb (hunk 14 above)
Comment (dsemenets) [reply recipes-bsp/xt-rpi-u-boot-scr/xt-rpi-u-boot-scr.bb:53]: Look at 2 lines above.
This script use mkimage to create bootscript. mkimage needs UBOOT_ARCH as argument.
I can take needed value manually or use kernel-arch

File: recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts (hunk 19 above)
Comment (dsemenets) [reply recipes-kernel/linux/files/bcm2712-raspberrypi5-domu.dts]: Comment added

**Review by lorc — APPROVED** [review]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Review by lorc — APPROVED** [review]

**Review by oleksiimoisieiev — APPROVED** [review]
